TalkAloud/TalkAloud/ViewControllers/RecordingViewController.swift

```diff
@@ -42,27 +43,36 @@ class RecordingViewController: UIViewController {
     @IBAction func recordButtonAction(_ sender: Any) {
         let audioState = AudioEngine.sharedInstance.audioState
         
-        switch audioState {
-        case .stopped:
-            guard let url = AudioManager.sharedInstance.createNewAudioRecording()?.url else { return }
-            AudioEngine.sharedInstance.setupRecorder(fileURL: url)
-            AudioEngine.sharedInstance.record()
-            recordButton.setImage(UIImage(named: "stopbutton"), for: .normal)
-            intializeRecordingTimer()
-            displayAudioVisualizer(audioState: AudioEngine.sharedInstance.audioState)
-            audioRecordingVisualizer.active = true
-            audioRecordingVisualizer.isHidden = false
-        case .recording:
-            AudioEngine.sharedInstance.stop()
-            resetView()
-            visualizerTimer?.invalidate()
-        case .paused:
-            print("Paused should never be happening")
-        case .playing:
-            print("Should never get here")
+        //        AudioEngine.sharedInstance.promptForMicrophonePermissions()
+        
+        AVAudioSession.sharedInstance().requestRecordPermission { (isGranted) in
```

Comment:
This logic could be encapsulated in the engine but leave it here for now. Just add a TODO.